packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -315,11 +328,24 @@ export abstract class FunctionBase extends Resource implements IFunction {
 
 export abstract class QualifiedFunctionBase extends FunctionBase {
   public abstract readonly lambda: IFunction;
+
+  /**
+   * The version or alias of this function
+   */
+  public abstract readonly qualifier: string;
   public readonly permissionsNode = this.node;
 
   public get latestVersion() {
     return this.lambda.latestVersion;
   }
+
+  public configureAsyncInvoke(options: EventInvokeConfigOptions): void {
+    new EventInvokeConfig(this, 'EventInvokeConfig', {
+      function: this.lambda,
+      qualifier: this.qualifier,
+      ...options
+    });
+  }
```

Comment:
```suggestion
  public configureAsyncInvoke(options: EventInvokeConfigOptions): void {
    if (this.node.tryFindChild('EventInvokeConfig') !== undefined) {
      throw new Error(`An EventInvokeConfig has already been configured for the qualified function ${this.functionName}:${this.qualifier}`);
    }

    new EventInvokeConfig(this, 'EventInvokeConfig', {
      function: this.lambda,
      qualifier: this.qualifier,
      ...options
    });
  }
```

and a corresponding unit test in `test.lambda.ts` that uses a private subclass of `QualifiedFunctionBase`.

Reply:
`this.qualifier` can be a token here, will adapt error message accordingly.

Reply:
will add unit tests for alias and version instead of having to fully define a subclass for `QualifiedFunctionBase` (more than 7 members must be fined...)

Reply:
i suggested a private subclass because I like to keep the unit tests for a specific code in its correspondingly named unit test file. Easy to discover the unit tests for that code branch, otherwise it's all over the place, and discovery is through test failure.

However, I'm not strongly opinionated on this so I'll let you decide.